packages/core/src/codewhisperer/service/securityScanHandler.ts

```diff
@@ -109,19 +111,34 @@ function mapRawToCodeScanIssue(issue: RawCodeScanIssue): CodeScanIssue {
     }
 }
 
-function mapToAggregatedList(codeScanIssueMap: Map<string, RawCodeScanIssue[]>, json: string) {
+function mapToAggregatedList(
+    codeScanIssueMap: Map<string, RawCodeScanIssue[]>,
+    json: string,
+    editor: vscode.TextEditor | undefined,
+    scope: CodeWhispererConstants.CodeAnalysisScope
+) {
     const codeScanIssues: RawCodeScanIssue[] = JSON.parse(json)
     codeScanIssues.forEach(issue => {
-        if (codeScanIssueMap.has(issue.filePath)) {
-            const list = codeScanIssueMap.get(issue.filePath)
-            if (list === undefined) {
-                codeScanIssueMap.set(issue.filePath, [issue])
+        let shouldShow = true
+
+        if (scope === CodeWhispererConstants.CodeAnalysisScope.FILE && editor) {
+            shouldShow = issue.codeSnippet.every(code => {
+                const line = editor.document.lineAt(code.number - 1)
+                return line?.text === code.content
+            })
+        }
+
+        if (shouldShow) {
+            const filePath = issue.filePath
+            if (codeScanIssueMap.has(filePath)) {
+                const issueList = codeScanIssueMap.get(filePath)
+                if (issueList) {
+                    issueList.push(issue)
+                    codeScanIssueMap.set(filePath, issueList)
+                }
```

Comment:
There's an else block missing, is that intentional?

```typescript
codeScanIssueMap.set(filePath, [issue])
```

Reply:
I think Both code snippet does the same functionality